source/material_model/melt_global.cc

```diff
@@ -0,0 +1,439 @@
+          eta_0                             = prm.get_double ("Reference shear viscosity");
+          xi_0                              = prm.get_double ("Reference bulk viscosity");
+          eta_f                             = prm.get_double ("Reference melt viscosity");
+          reference_permeability            = prm.get_double ("Reference permeability");
+          thermal_viscosity_exponent        = prm.get_double ("Thermal viscosity exponent");
+          thermal_bulk_viscosity_exponent   = prm.get_double ("Thermal bulk viscosity exponent");
+          thermal_conductivity              = prm.get_double ("Thermal conductivity");
+          reference_specific_heat           = prm.get_double ("Reference specific heat");
+          thermal_expansivity               = prm.get_double ("Thermal expansion coefficient");
+          alpha_phi                         = prm.get_double ("Exponential melt weakening factor");
+          depletion_density_change          = prm.get_double ("Depletion density change");
+          surface_solidus                   = prm.get_double ("Surface solidus");
+          depletion_solidus_change          = prm.get_double ("Depletion solidus change");
+          pressure_solidus_change           = prm.get_double ("Pressure solidus change");
+          peridotite_melting_entropy_change = prm.get_double ("Peridotite melting entropy change");
+          compressibility                   = prm.get_double ("Solid compressibility");
+          melt_compressibility              = prm.get_double ("Melt compressibility");
+          include_melting_and_freezing      = prm.get_bool ("Include melting and freezing");
+
+          if (thermal_viscosity_exponent!=0.0 && reference_T == 0.0)
+            AssertThrow(false, ExcMessage("Error: Material model Melt simple with Thermal viscosity exponent can not have reference_T=0."));
+
+
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+  }
+}
+
+// explicit instantiations
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    ASPECT_REGISTER_MATERIAL_MODEL(MeltGlobal,
+                                   "melt global",
+                                   "A material model that implements a simple formulation of the "
+                                   "material parameters required for the modelling of melt transport, "
+                                   "including a source term for the porosity according to a simplified "
+                                   "linear melting model:\n"
+                                   "$\\phi_\\text{equilibrium} = \\frac{T-T_\\text{sol}}{T_\\text{liq}-T_\\text{sol}}$\n"
+                                   "with "
+                                   "$T_\\text{sol} = T_\\text{sol,0} + \\Delta T_p \\, p + \\Delta T_c \\, C$ \n"
+                                   "$T_\\text{liq} = T_\\text{sol}  + \\Delta T_\\text{sol-liq}$.")
```

Comment:
Want to add a reference to a publication here? You can add a `\\cite{abc}` if you add `abc` to `doc/manual/manual.bib`.